scripts/build_containers.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import os
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from pathlib import Path
+from typing import List
+
+from jsonargparse import ArgumentParser
+from spython.main import Client
+
+# Define the directory where the projects are located
+BASE_DIR: Path = Path(__file__).resolve().parent.parent / "projects"
+
+# List of all available project names
+PROJECTS: List[str] = [x.name for x in BASE_DIR.iterdir() if x.is_dir()]
+
+
+def build_container(project_name: str, container_root: Path) -> str:
+    project_path: Path = BASE_DIR / project_name
+    container_path: Path = container_root / f"{project_name}.sif"
+
+    if not container_path.exists():
```

Comment:
Should this be checking whether the definition file exists?

Reply:
This is just skipping over the projects that don't have apptainer.defs

Reply:
It looks like it's skipping over the ones without an already existing `.sif` file, so it wouldn't do anything if you hadn't built any containers in the first place, or had deleted one of them.

Reply:
Ooh. yeah I see, checked for wrong variable here.

Reply:
done